R/as_curve_data.R

```diff
@@ -0,0 +1,60 @@
+#' Load antibody decay curve parameter
+#'
+#' @param data a [data.frame()] or [tibble::tbl_df]
+#' @param antigen_isos [character()] vector of antigen isotypes to be used in analyses
+#' @returns a `curve_data` object (a [tibble::tbl_df] with extra attribute `antigen_isos`)
+#' @export
+#' @examples
+#' curve_data <- load_curve_params("https://osf.io/download/rtw5k/")
+#'
+#' print(curve_data)
+#'
+as_curve_data <- function(data, antigen_isos = NULL) {
+
+  # convert data to a tibble data type
+  tibble_data <-
+    data %>%
+    tibble::as_tibble()
+
+  # convert data to curve class
+  converted_data <-
+    tibble_data %>% curve_converter()
+
+  return(curve_data)
+}
+
+curve_converter <- function(data, antigen_isos = NULL) {
```

Comment:
decomposing out `curve_converter()` seems like unnecessary complexity; there's only two commands left in `as_curve_data()`. is there a reason for doing it this way?